### DISCUSSION TITLE
How to stop validation/re-rendering in certain scenarios? (e.g. don't revalidate when search params change)

### DISCUSSION THREAD

**Discussion by jvllmr** [TanStack/router#1833] [Q&A]
I am migrating from `react-router-dom` to `@tanstack/router` and I'm facing a critical issue.

The application has SVG Viewers within multiple routes in the application and they update the search params with x, y, z coordinates when panning/zooming the viewer. When the search gets updates, the whole route re-validates and re-renders, resulting in a lot of stuttering.

With react-router I managed to solve this via the `shouldRevalidate` option (https://reactrouter.com/en/main/route/should-revalidate). Can I achieve the same in `@tanstack/router`? I tried using the `shouldReload` option, but it didn't work the same.

**Comment by jvllmr** [comment]
I think PR #1556  is somewhat related to this.

**Comment by lazybensch** [comment]
@jvllmr did you find a way to prevent tanstack from rerendering when search params change?

**Reply by jvllmr** [reply]
No I did not. For now I worked around  this issue by only navigating when the user did not move the svg viewer for 500 milliseconds.

**Comment by zotodev** [comment]
if you are using useSearch hook you might have to include select inside it to just subscribe to the params you need rerender page for. For other params for which you dont want to rerender on change, you can use useState(router.state.location.search) instead of useSearch.
